### PR TITLE
[ROCKETMQ-39]  avoid duplicated codes.

### PR DESCRIPTION
**Pull request by wu-sheng**
ref issue: https://issues.apache.org/jira/browse/ROCKETMQ-39

### CODE REVIEW

**Comment by lizhanhui** [comment]
The extracted shutdown hook misses required license header and causes check-style failure. Please fix it and double check if there are more similar code style issues.

You may refer [here](http://rocketmq.incubator.apache.org/docs/code-guidelines/) in terms of code style.

**Comment by vongosling** [comment]
@wu-sheng Could you modify your PR , following our codestyle and PR guide[1] :-)

[1]http://rocketmq.incubator.apache.org/docs/code-guidelines/

**Comment by wu-sheng** [comment]
@vongosling , fixed and done. Hoping `style-check` works well next time. 😭

File: filtersrv/src/main/java/org/apache/rocketmq/filtersrv/FiltersrvStartup.java [hunk 1]
```diff
@@ -135,27 +137,17 @@ public static FiltersrvController createController(String[] args) {
...
-                        log.info("shutdown hook was invoked, " + this.shutdownTimes.incrementAndGet());
-                        if (!this.hasShutdown) {
-                            this.hasShutdown = true;
-                            long begineTime = System.currentTimeMillis();
-                            controller.shutdown();
-                            long consumingTimeTotal = System.currentTimeMillis() - begineTime;
-                            log.info("shutdown hook over, consuming time total(ms): " + consumingTimeTotal);
-                        }
-                    }
+                public Object call() throws Exception {
```
Comment (WillemJiang) [inline filtersrv/src/main/java/org/apache/rocketmq/filtersrv/FiltersrvStartup.java]: If you don't want to return any thing here , you could define the method just like this.
    public Void call() throws Exception {
    }

Comment (wu-sheng) [inline filtersrv/src/main/java/org/apache/rocketmq/filtersrv/FiltersrvStartup.java]: @WillemJiang done.

**Comment by zhouxinyu** [comment]
It could be better if we add some unit tests for this PR and please remove the author info.

File: filtersrv/src/main/java/org/apache/rocketmq/filtersrv/FiltersrvStartup.java [hunk 2]
```diff
@@ -139,7 +139,7 @@ public static FiltersrvController createController(String[] args) {
 
             Runtime.getRuntime().addShutdownHook(new ShutdownHookThread(log, new Callable() {
                 @Override
-                public Object call() throws Exception {
+                public Void call() throws Exception {
                     controller.shutdown();
                     return null;
```
Comment (WillemJiang) [inline filtersrv/src/main/java/org/apache/rocketmq/filtersrv/FiltersrvStartup.java]: When use the Void as the return signature, you don't need to use return in the function.  
Please remove the line of 144.

File: namesrv/src/main/java/org/apache/rocketmq/namesrv/NamesrvStartup.java [hunk 3]
```diff
@@ -123,7 +123,7 @@ public static NamesrvController main0(String[] args) {
 
             Runtime.getRuntime().addShutdownHook(new ShutdownHookThread(log, new Callable() {
                 @Override
-                public Object call() throws Exception {
+                public Void call() throws Exception {
                     controller.shutdown();
                     return null;
```
Comment (WillemJiang) [inline namesrv/src/main/java/org/apache/rocketmq/namesrv/NamesrvStartup.java]: This line should be removed.

File: filtersrv/src/main/java/org/apache/rocketmq/filtersrv/FiltersrvStartup.java (hunk 2 above)
Comment (wu-sheng) [inline filtersrv/src/main/java/org/apache/rocketmq/filtersrv/FiltersrvStartup.java]: @WillemJiang, Are you sure about no need `return null`? I have run a `mvn package` on local, and you can see, `return statement` is required.

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.5.1:compile (default-compile) on project rocketmq-namesrv: Compilation failure
[ERROR] /Users/wusheng/Documents/code/github/incubator-rocketmq-fork/namesrv/src/main/java/org/apache/rocketmq/namesrv/NamesrvStartup.java:[128,17] 缺少返回语句
[ERROR] -> [Help 1]
```

Comment (WillemJiang) [inline filtersrv/src/main/java/org/apache/rocketmq/filtersrv/FiltersrvStartup.java]: @wu-sheng ,  oh, it's my mistake, I just wrote a code and the return statement need to be kept.
As I'm not a big fan of null,  I always want to kill it if possible.

Comment (wu-sheng) [inline filtersrv/src/main/java/org/apache/rocketmq/filtersrv/FiltersrvStartup.java]: +1 on `kill return`, sadly, it's not working.

**Comment by dongeforever** [comment]
@wu-sheng this PR has been merged. you may close the PR.

**Pull request closed without merging**